### PR TITLE
For #417: ItemListView test with authentication.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #417:  `ItemListView`  test with authentication.
- Corrected the `ItemListView` session problem using other logic to create fake objects
- Left puzzle for correcting `ItemListView` behavior

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#507](https://github.com/timelesslounge/timelessis/issues/507) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#507](https://github.com/timelesslounge/timelessis/issues/507) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/587ff066-6c71-4022-ad43-2ff1618ef205); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/view/items/test_items.py [hunk 1]
```diff
@@ -8,27 +8,34 @@
...
-    employee = factories.EmployeeFactory(company=company)
-    factories.ItemFactory(employee_id=employee.id, company=company)
+    employee = factories.EmployeeFactory()
+    factories.ItemFactory(name="1", )
+    factories.ItemFactory(name="2")
+    factories.ItemFactory(name="3")
+    factories.ItemFactory(name="4")
+    with client.session_transaction() as session:
+        session["user_id"] = employee.id
+    flask.g.user = employee
```
Comment (amihaiemil) [inline tests/view/items/test_items.py]: @paulodamaso shouldn't ``flask.g.user`` be initialized before creating the session?

File: tests/view/items/test_items.py [hunk 2]
```diff
@@ -8,27 +8,34 @@
...
 
-@pytest.mark.skip(reason="Correct the factory bug")
+@pytest.mark.skip(reason="Correct the ItemListView bug")
 def test_list(client):
     """ Test list is okay """
-    company = factories.CompanyFactory()
-    employee = factories.EmployeeFactory(company=company)
-    factories.ItemFactory(employee_id=employee.id, company=company)
+    employee = factories.EmployeeFactory()
+    factories.ItemFactory(name="1", )
```
Comment (amihaiemil) [inline tests/view/items/test_items.py]: @paulodamaso can you remove the trailing ``, ``?

File: tests/view/items/test_items.py [hunk 3]
```diff
@@ -8,27 +8,34 @@
...
+    factories.ItemFactory(name="2")
+    factories.ItemFactory(name="3")
+    factories.ItemFactory(name="4")
```
Comment (amihaiemil) [inline tests/view/items/test_items.py]: @paulodamaso I think just 3 items are enough, 4 seem too much :D

File: tests/view/items/test_items.py [hunk 4]
```diff
@@ -8,27 +8,34 @@
...
     response = client.get("/items/")
-    assert "<article class=\"item\"><header><div><h1>1</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>2</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>3</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>4</h1></div>" in response.data
+    print(response.data)
+    assert b"<article class=\"item\"><header><div><h1>1</h1></div>" in response.data
+    assert b"<article class=\"item\"><header><div><h1>2</h1></div>" in response.data
+    assert b"<article class=\"item\"><header><div><h1>3</h1></div>" in response.data
+    assert b"<article class=\"item\"><header><div><h1>4</h1></div>" in response.data
```
Comment (amihaiemil) [inline tests/view/items/test_items.py]: @paulodamaso can you also assert that these items really belong to the authenticated employee?

Comment (paulodamaso) [reply tests/view/items/test_items.py]: @amihaiemil we're testing the screen here, and we don't have this field available onscreen unfortunately

**Comment by paulodamaso** [comment]
@amihaiemil Pushed the corrections, please take a look

**Comment by amihaiemil** [comment]
@paulodamaso nice, thanks :)

**Comment by amihaiemil** [comment]
@rultor good to merge

File: tests/view/items/test_items.py [hunk 5]
```diff
@@ -8,27 +8,32 @@
...
+    factories.ItemFactory(name="3")
+    flask.g.user = employee
+    with client.session_transaction() as session:
+        session["user_id"] = employee.id
     response = client.get("/items/")
-    assert "<article class=\"item\"><header><div><h1>1</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>2</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>3</h1></div>" in response.data
-    assert "<article class=\"item\"><header><div><h1>4</h1></div>" in response.data
+    print(response.data)
```
Comment (emilianodellacasa) [inline tests/view/items/test_items.py]: @paulodamaso please remove this pribt

**Comment by paulodamaso** [comment]
@emilianodellacasa Done, please take a look

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#507](https://github.com/timelesslounge/timelessis/issues/507) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by 0crat** [comment]
The job [#507](https://github.com/timelesslounge/timelessis/issues/507) is now out of scope

**Comment by amihaiemil** [comment]
@ypshenychka I confirn

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by amihaiemil** [comment]
@ypshenychka ping :)

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**